File: TCP_Module/src/main/java/com/common/crypto/AesGcmChunk.java
package com.common.crypto;

import java.util.Arrays;
import java.util.Objects;

public final class AesGcmChunk {
    private final byte[] nonce;
    private final byte[] ciphertext;
    private final byte[] tag;

    public AesGcmChunk(byte[] nonce, byte[] ciphertext, byte[] tag) {
        this.nonce = nonce;
        this.ciphertext = ciphertext;
        this.tag = tag;
    }


    public byte[] nonce() {
        return nonce;
    }

    public byte[] ciphertext() {
        return ciphertext;
    }

    public byte[] tag() {
        return tag;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AesGcmChunk that)) {
            return false;
        }
        return Objects.equals(nonce, that.nonce)
                && Objects.equals(ciphertext, that.ciphertext)
                && Objects.equals(tag, that.tag);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nonce, ciphertext, tag);
    }//计算哈希值

    @Override
    public String toString() {
        return "AesGcmChunk[nonce=" + nonce
                + ", ciphertext=" + ciphertext
                + ", tag=" + tag + "]";
    }
}

File: Android-FileSecurityTool-OfflineEncryptDecrypt/app/src/main/kotlin/com/app/crypto/AesGcmChunk.kt
package com.filesecuritytool.android.crypto

/**
 * Port of AesGcmChunk.java — holds the three components of an AES-256-GCM encryption output.
 * Nonce is 12 bytes, ciphertext is variable length, tag is 16 bytes.
 */
data class AesGcmChunk(
    val nonce: ByteArray,
    val ciphertext: ByteArray,
    val tag: ByteArray
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is AesGcmChunk) return false
        return nonce.contentEquals(other.nonce) &&
                ciphertext.contentEquals(other.ciphertext) &&
                tag.contentEquals(other.tag)
    }

    override fun hashCode(): Int {
        var result = nonce.contentHashCode()
        result = 31 * result + ciphertext.contentHashCode()
        result = 31 * result + tag.contentHashCode()
        return result
    }

    override fun toString(): String {
        return "AesGcmChunk(nonce=${nonce.joinToString("") { "%02x".format(it) }.take(12)}..., " +
                "ciphertext=${ciphertext.size}b, tag=${tag.joinToString("") { "%02x".format(it) }.take(12)}...)"
    }
}
